src/modules/smsops/smsops_impl.c

```diff
@@ -618,6 +627,7 @@ int decode_3gpp_sms(struct sip_msg *msg) {
 					// Coding: 7 Bit
 					if (rp_data->pdu.coding == 0x00) {
 						// We don't care about the extra used bytes here.
+						rp_data->pdu.payload.sm.len = len;
 						rp_data->pdu.payload.sm.len = gsm_to_ascii(&body.s[p], len, rp_data->pdu.payload.sm, fill_bits);
```

Comment:
Above are now two assignments one after the other to `rp_data->pdu.payload.sm.len`, is it `rp_data->pdu.payload.sm.len` used inside `gsm_to_ascii()`? Asking because len is also provided as parameter to `gsm_to_ascii()`.

Reply:
Yes "rp_data->pdu.payload.sm" is used in the gsm_to_ascii(...). You can see that sms.len is used in a few "if" statements.

Reply:
OK, I will merge it. Somehow confusing is that `len` is also explicitly passed as a parameter to that function.